apps/frontend/src/libs/components/popup/popup.tsx

```diff
@@ -0,0 +1,60 @@
+import {
+	useCallback,
+	useOnClickOutside,
+	useRef,
+	useState,
+} from "~/libs/hooks/hooks.js";
+
+import styles from "./styles.module.css";
+
+const ESCAPE_KEY = "Escape";
+
+type Properties = {
+	children: React.ReactNode;
+	content: React.ReactNode;
+};
+
+const Popup = ({ children, content }: Properties): JSX.Element => {
```

Comment:
we have a modal component for it

Reply:
But they are not the same, modal is positioned in center of the screen, while pop-ups will be positioned relative to buttons that trigger their button. Also, modal has a backdrop and blocks interactions with everything, while pop-up does not. And as for modal used dialog element from html, we will not be able to use it as pop-up. I think it's the best to use separate components for modal and pop-up. However, I can reuse outside click hook from that PR, I'm just waiting for it to be merged.
<img width="637" alt="Screenshot 2024-08-27 at 14 34 00" src="https://github.com/user-attachments/assets/41ad5fdc-1f52-4fe6-b72c-a93af74c8fd3">
<img width="364" alt="Screenshot 2024-08-27 at 14 41 29" src="https://github.com/user-attachments/assets/fa591fe5-8dd9-4605-bacf-08d8d9db5dd9">